### PR TITLE
Fix equals and hashcode for SingleValueQuery.LuceneQuery

### PR DESCRIPTION
**Pull request by iverase**
Fixes the implementation of equals and hashcode so the lucene queries can be cached.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQuery.java [hunk 1]
```diff
@@ -267,14 +268,12 @@ public boolean equals(Object obj) {
...
-            return next.equals(other.next)
-                && fieldData.getFieldName().equals(other.fieldData.getFieldName())
-                && warnings.equals(other.warnings);
+            return next.equals(other.next) && fieldData.getFieldName().equals(other.fieldData.getFieldName());
         }
 
         @Override
         public int hashCode() {
-            return Objects.hash(classHash(), next, fieldData, warnings);
+            return Objects.hash(classHash(), next, fieldData.getFieldName());
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQuery.java:276]: 👍

**Review by bpintea — APPROVED** [review]

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Comment by iverase** [comment]
Thank you @bpintea & @ChrisHegarty!

**Comment by costin** [comment]
Worth backporting the bug to 8.14.x

**Comment by iverase** [comment]
backporting in progress, lazy bot did not help.

**Pull request merged**